### PR TITLE
[FLAVA]Move image transform to core

### PR DESCRIPTION
**Pull request by ankitade**
Move FLAVA image transform to core without any logical changes. Add unit test and typing info

Test plan
pytest

Sanity checks
python -m flava.finetune config=flava/configs/finetuning/qnli.yaml training.lightning.accelerator=cpu training.lightning.gpus=0 training.lightning.strategy=null

python -m flava.train config=flava/configs/pretraining/debug.yaml training.lightning.accelerator=cpu training.lightning.gpus=0 training.lightning.strategy=null

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #71

Differential Revision: [D39049377](https://our.internmc.facebook.com/intern/diff/D39049377)

### CODE REVIEW

**Review by langong347 — COMMENTED** [review]
Would be good to add docstrings to explain the major components.

**Comment by ankitade** [comment]
@deankita has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39049377).

File: .flake8 [hunk 1]
```diff
@@ -6,7 +6,7 @@ max-line-length = 120
 # E501 is not flexible enough, we're using B950 instead
 # N812 ignored because import torch.nn.functional as F is PyTorch convention
 ignore =
```
Comment (langong347) [inline .flake8:8]: Are the ignores removed by accident?

File: test/transforms/test_image_masking_transform.py [hunk 2]
```diff
@@ -4,21 +4,19 @@
...
+from test.test_utils import assert_expected, set_rng_seed
+from torchmultimodal.transforms.flava_transform import MaskedImageModelingTransform
 from torchvision import transforms
 
 
-class TestImageMaskingTransforms(unittest.TestCase):
-    def setUp(self):
-        self.test_image = get_asset_path("grace_hopper_517x606.jpg")
+class TestImageMaskingTransforms:
+    @pytest.fixture
```
Comment (langong347) [inline test/transforms/test_image_masking_transform.py:15]: nit: You can set `autouse=True` for seed setting such that you won't need to request it for any test/fixture that depends on it.

File: torchmultimodal/transforms/flava_transform.py [hunk 3]
```diff
@@ -200,9 +204,30 @@ def __call__(self, img):
 
 
 class MaskedImageModelingTransform:
+    """Masked image modeling transform
+    Randomly masks patches in an image based on scheme from Beit
+    https://arxiv.org/pdf/2106.08254.pdf
+
+    Args:
+        is_train (bool): whether transform is applied during training or not
```
Comment (langong347) [inline torchmultimodal/transforms/flava_transform.py:212]: For arguments with default values, it's typical to add a statement at the end of each docstring. For example:

```
is_train (bool): whether transform is applied during training or not. Defaults to ``True``.
```

**Comment by ankitade** [comment]
@deankita has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39049377).

File: torchmultimodal/transforms/flava_transform.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+
+def map_pixels(x: torch.Tensor) -> torch.Tensor:
+    if x.dtype != torch.float:
+        raise ValueError("expected input to have type float")
+
+    return (1 - 2 * LOGIT_LAPLACE_EPS) * x + LOGIT_LAPLACE_EPS
+
+
+class ImageMaskingGenerator:
```
[the remark this answers is not in the document]
Comment (ankitade) [reply torchmultimodal/transforms/flava_transform.py:31]: i think its ok to not have for classes not meant for public use. but can add in follow up

**Review by ebsmothers — APPROVED** [review]

**Comment by ankitade** [comment]
@deankita has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39049377).
